src/lib.rs

```diff
@@ -1015,6 +1015,100 @@ impl Rng for ThreadRng {
     }
 }
 
+/// An RNG provided specifically for seeding PRNG's. It uses the interface for
+/// random numbers provided by the operating system (`OsRng`). If that returns
+/// an error, it will fall back to the `JitterRng` entropy collector.
+/// Occasionally it will then check if `OsRng` is still not available.
+#[derive(Debug)]
+pub struct EntropySource {
+    rng: EntropySourceInner,
+    counter: u32,
+}
+
+#[derive(Debug)]
+enum EntropySourceInner {
+    Os(OsRng),
+    Jitter(JitterRng),
+}
+
+impl EntropySource {
+    pub fn new() -> Result<Self, Error> {
```

Comment:
I think it's worth making this infallible and delaying errors so that `X::from_rng(EntropySource::new())` works. Since performance isn't very important here we could potentially just try all sources each call, though that's not optimal.

Reply:
`JitterRng` has a pretty high set-up cost, because it runs the timer for every initialization.

Do you mean to add something like a third variant to `EntropySourceInner` that none are available, with the `Error`?

Reply:
`X::from_rng(EntropySource::new())` works really nice! It will take me until Wednesday or Thursday before I can update this PR.